### DISCUSSION TITLE
Can a lineart model be added to SDXL?

### DISCUSSION THREAD

**Discussion by mald55** [lllyasviel/Fooocus#2137] [Ideas]
Hi,

Cany any of the lineart models be added to Fooocus? I feel they are great when making sketches into life like images.  The inverted line art model is amazing on SD 1.5

**Comment by eddyizm** [comment]
Which one? if the model is SDXL you can download and use it but if not then you can't. Fooocus doesn't control the models per say unless I am misunderstanding your question.

**Comment by mashb1t** [comment]
@mald55 please check out one of the various line art or pencil drawing styles Fooocus already offers. Feel free to download and put any SDXL model of your liking to the models > checkpoints folder

**Reply by mald55** [reply]
So a few questions:

1-) I have several SDXL checkpoints inside the checkpoint folder, if I were to drop a model from [here ](https://huggingface.co/lllyasviel/sd_control_collection/tree/main) - (let's say open pose for example), how would I select it within the GUI when I input an image as a prompt.  I am coming from automatic1111 when you select controlnet models by hand in a different section of the GUI- apart from the checkpoint selection.

![controlnet fooocus](https://github.com/lllyasviel/Fooocus/assets/45581991/cd801689-67bf-4ba7-8b40-1f38cf2743b8)

Also, Is PyraCanny and CPDS tied to specific models?

2-) To my understanding the styles offered by fooocus were just baked keywords that get added to your prompt, and not controlnet models? When I select FooocusV2 for example I just see more keywords being added to my prompt in the command line, how does that equate to a proper lineart controlnet model? _**My goal is to turn a sketch/black and white drawing into something more life like.**_ 

3-) Lastly, where can I find a proper lineart model similar to [this ](https://huggingface.co/lllyasviel/control_v11p_sd15_lineart) for SDXL, as far I know there isn't one for SDXL? I have looked everywhere and haven't seen anything that comes close to it, am I wrong?

Sorry for the long reply, just started using the vanilla Fooocus a couple days back, after using FooocusMRE/Automatic1111 for the last 4 months.

**Reply by mashb1t** [reply]
1) Fooocus does not support other control types than the 4 shown in the image, so also no lineart controlnet. PyraCanny and CPDS are not tied to specific checkpoints but are preprocessors/controlnets themselves.
2) Styles are keywords in the prompt, true. I must have misunderstood your question, sorry. You can play around with vary to make the more lifelike.
3) See 1), there is currently no support for this OOTB, feel free to migrate it ovr from fooocus-control if it exists there https://github.com/fenneishi/Fooocus-ControlNet-SDXL

**Comment by mald55** [comment]
> @mald55 please check out one of the various line art or pencil drawing styles Fooocus already offers. Feel free to download and put any SDXL model of your liking to the models > checkpoints folder



> 1. Fooocus does not support other control types than the 4 shown in the image, so also no lineart controlnet. PyraCanny and CPDS are not tied to specific checkpoints but are preprocessors/controlnets themselves.
> 2. Styles are keywords in the prompt, true. I must have misunderstood your question, sorry. You can play around with vary to make the more lifelike.
> 3. See 1), there is currently no support for this OOTB, feel free to migrate it ovr from fooocus-control if it exists there https://github.com/fenneishi/Fooocus-ControlNet-SDXL

Perfect, I didn't know that fork existed, I'll take a look.

Btw are there plants to add more types controlnet such as pose, depth, recolor and sketch to Fooocus? If not, why?

**Reply by mashb1t** [reply]
I am not aware of any plans and have now posted a question targeted towards the maintainer: https://github.com/lllyasviel/Fooocus/discussions/2154
